sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/AbstractSpannerFn.java

```diff
@@ -36,7 +39,16 @@
   @Setup
   public void setup() throws Exception {
     SpannerConfig spannerConfig = getSpannerConfig();
-    SpannerOptions options = spannerConfig.buildSpannerOptions();
+    SpannerOptions.Builder builder = SpannerOptions.newBuilder();
+    if (spannerConfig.getProjectId() != null) {
+      builder.setProjectId(spannerConfig.getProjectId().get());
+    }
+    if (spannerConfig.getServiceFactory() != null) {
+      builder.setServiceFactory(spannerConfig.getServiceFactory());
+    }
+    ReleaseInfo releaseInfo = ReleaseInfo.getReleaseInfo();
+    builder.setUserAgentPrefix(USER_AGENT_PREFIX + "." + releaseInfo.getVersion());
```

Comment:
I would not be surprised if at some point ReleaseInfo.getName() may return a non-hardcoded value; e.g. if somebody is using Beam via https://github.com/spotify/scio/, I think it'd be nice if ReleaseInfo said that this is scio, even though currently it doesn't say that. There doesn't seem to any downside to using ReleaseInfo.getName() instead of hardcoding "Apache_Beam_Java" yourself, is there?